src/main/java/cn/nukkit/block/BlockClosedEyeblossom.java

```diff
@@ -19,4 +20,33 @@ public BlockClosedEyeblossom(BlockState blockstate) {
         super(blockstate);
     }
 
+    @Override
+    public int onUpdate(int type) {
+        if(type == Level.BLOCK_UPDATE_RANDOM || type == Level.BLOCK_UPDATE_SCHEDULED) {
+            boolean changed = false;
+            if(getId().equals(CLOSED_EYEBLOSSOM)) {
+                if(level.isNight()) changed = true;
+            } else if(getId().equals(OPEN_EYEBLOSSOM)) {
+                if(level.isDay()) changed = true;
+            }
```

Comment:
This can be simplified:

```java
boolean changed = getId().equals(CLOSED_EYEBLOSSOM) && level.isNight() || getId().equals(OPEN_EYEBLOSSOM) && level.isDay();
```